magento/magento2
# Backport MAGETWO-70726

### Description
This is a backport of branch [MAGETWO-70726](https://github.com/magento/magento2/commit/e71416c23435e3f22dbd472a3510326c0f2dd14e)
Fixes 404 issues when switching stores on sub-category level.


### Fixed Issues (if relevant)

1. magento/magento2#5416:  Store View Language switch leads to 404 on some cases


### Manual testing scenarios

1. Create two stores
2. Create categories with sub categories.
3. Make sure the url-keys are different on each category
4. Navigate to a sub category
5. Switch stores.

### Contribution checklist
